src/wallet/wallet.cpp

```diff
@@ -4281,11 +4281,11 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(const std::string& name,
         //We can't rescan beyond non-pruned blocks, stop and throw an error
         //this might happen if a user uses an old wallet within a pruned node
         // or if he ran -disablewallet for a longer time, then decided to re-enable
-        if (fPruneMode)
-        {
+        if (fPruneMode) {
             CBlockIndex *block = chainActive.Tip();
-            while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA) && block->pprev->nTx > 0 && pindexRescan != block)
+            while (block && block->pprev && !IsBlockPruned(block->pprev) && pindexRescan != block) {
```

Comment:
nit, could drop first condition `block && `.

Reply:
`block` is updated inside the loop, so that sounds like it could be unsafe. (That said, not sure if there is ever a case where a `CBlockIndex` goes from being set to being unset, except for genesis block...)

Reply:
`block` is updated to `block->pprev` which is already tested.

Reply:
Ah, yes you're right. Done.